src/File/RenameUploadFactory.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter\File;
+
+use Laminas\ServiceManager\Factory\FactoryInterface;
+use Psr\Container\ContainerInterface;
+
+final class RenameUploadFactory implements FactoryInterface
+{
+    public function __invoke(
+        ContainerInterface $container,
+        string $requestedName,
+        array|null $options = null,
+    ): RenameUpload {
+        /**
+         * @psalm-suppress MixedArgumentTypeCoercion Runtime validation of options is not worth it.
+         */
+        return new RenameUpload($options ?? [], new MoveUploadedFile());
```

Comment:
Have I overlooked something or why is the factory needed?

Reply:
Arguably, we could inline the `MoveUploadedFile` class. It was extracted so that we could stub PHP's `move_uploaded_file` function, but now the default implementation checks `is_uploaded_file` _(and runs `rename` if its not)_, it feels less important that its a stand-alone collaborator. The only other benefit of keeping it external is that it now also runs an error handler to upcast PHP errors/warnings to exceptions, but we could also just `new` it in the constructor and drop the factory completely.

What do you think?

Reply:
> …but we could also just `new` it in the constructor and drop the factory completely.

It is already there:

```php
$this->moveUploadedFile = $moveUploadedFile ?? new MoveUploadedFile();
```

https://github.com/laminas/laminas-filter/pull/239/files#diff-94631e5208b9361631c5713c4af740513be7608554440d9130f6e8f633498bd5R62

Reply:
Thanks @froschdesign - I've dropped the factory now 👍